core_impl/src/main/java/io/opencensus/implcore/tags/CurrentTaggingState.java

```diff
@@ -29,13 +31,25 @@
  */
 @ThreadSafe
 public final class CurrentTaggingState {
-  private volatile TaggingState currentState = TaggingState.ENABLED;
 
-  public TaggingState get() {
+  @GuardedBy("this")
+  private TaggingState currentState = TaggingState.ENABLED;
+
+  @GuardedBy("this")
+  private boolean isRead;
+
+  public synchronized TaggingState get() {
+    isRead = true;
+    return getInternal();
+  }
+
+  public synchronized TaggingState getInternal() {
```

Comment:
This can be package-private.

Reply:
This method is called from TagContextBinarySerializerImpl, which is in a different package.